src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_Conditional.cs

```diff
@@ -690,6 +687,29 @@ class Bar
                 Diagnostic(ErrorCode.ERR_BadAttributeArgument, "Goo.M"));
         }
 
+        [Fact]
+        public void ConditionalAttributeArgument_Null()
+        {
+            var source =
+@"using System.Diagnostics;
+class Program
+{
+    [Conditional(""A"")]
```

Comment:
Can Conditional be applied to a lambda? (I'd think it should not)
We should review other attributes listed in `SourceMethodSymbolWithAttributes.DecodeWellKnownAttributeAppliedToMethod` as well.

Reply:
`ConditionalAttribute` and `ObsoleteAttribute` are not handled. Let me know if you think there is a scenario with either that needs to be addressed.

The other attributes should be handled.

Reply:
- ObsoleteAttribute: I agree (ie. don't care). But we should have a test
- ConditionalAttribute: I think that should be disallowed (as it is on local functions)

---
In reply to: [612020197](https://github.com/dotnet/roslyn/pull/52178#discussion_r612020197) [](ancestors = 612020197)

Reply:
I was incorrect. ConditionalAttribute is allowed on local functions. We should do the same.
[Example](https://sharplab.io/#v2:EYLgtghgzgLgpgJwD4GIB2BXANliwtwAEcaeBAsAFAACATAIxXUAMh19AdACICWEA5mgD2sHgGMoHAMJCAJnACCpLAE8oPKAG4mtNvQDsVAN5VCZvQDY2AFkIBZCDzQAKAJSnzJyuZ97dEekIAXkIsITEILG1vXzMwiKw3aNizagBWAB4hYAArODEYAH4APkIIXRD4yNpklI9YgG1qfRB7CBVgOAA5bCwAXUImzl4BYVEJaSE0WR4YHinI5wAiLgBRACEAVQBxJdcB6lsqxNdCI0IAX3rfJpbCLqEYHpwB7LyCwtDw6rdg0swcLUzFdKBcgA)

---
In reply to: [612132673](https://github.com/dotnet/roslyn/pull/52178#discussion_r612132673) [](ancestors = 612132673,612020197)